identity-iota-core/src/document/iota_document.rs

```diff
@@ -293,43 +293,24 @@ impl IotaDocument {
     Ok(self.document.detach_method_relationship(did_url, relationship)?)
   }
 
-  /// Returns the first [`IotaVerificationMethod`] with an `id` property
-  /// matching the provided `query`.
-  pub fn resolve_method<'query, Q>(&self, query: Q) -> Option<&IotaVerificationMethod>
+  /// Returns the first [`IotaVerificationMethod`] with an `id` property matching the
+  /// provided `query` and the verification relationship specified by `scope` if present.
+  pub fn resolve_method<'query, Q>(&self, query: Q, scope: Option<MethodScope>) -> Option<&IotaVerificationMethod>
   where
     Q: Into<DIDUrlQuery<'query>>,
   {
-    self.document.resolve_method(query)
-  }
-
-  /// Returns the first [`IotaVerificationMethod`] with an `id` property
-  /// matching the provided `query`.
-  ///
-  /// # Errors
-  ///
-  /// Fails if no matching verification [`IotaVerificationMethod`] is found.
-  pub fn try_resolve_method<'query, Q>(&self, query: Q) -> Result<&IotaVerificationMethod>
-  where
-    Q: Into<DIDUrlQuery<'query>>,
-  {
-    Ok(self.document.try_resolve_method(query)?)
+    self.document.resolve_method(query, scope)
   }
 
   #[doc(hidden)]
   pub fn try_resolve_method_mut<'query, Q>(&mut self, query: Q) -> Result<&mut IotaVerificationMethod>
   where
     Q: Into<DIDUrlQuery<'query>>,
   {
-    self.document.try_resolve_method_mut(query).map_err(Into::into)
-  }
-
-  /// Returns the first [`IotaVerificationMethod`] with an `id` property matching the provided `query`
-  /// and the verification relationship specified by `scope`.
-  pub fn resolve_method_with_scope<'query, Q>(&self, query: Q, scope: MethodScope) -> Option<&IotaVerificationMethod>
-  where
-    Q: Into<DIDUrlQuery<'query>>,
-  {
-    self.document.resolve_method_with_scope(query, scope)
+    self
+      .document
+      .resolve_method_mut(query)
+      .ok_or(Error::InvalidDoc(identity_did::Error::MethodNotFound))
```

Comment:
`IotaDocument::try_resolve_method_mut` should be renamed `resolve_method_mut` and return an `Option`, same as `CoreDocument`.